File: Projet_LO52/lo52_project_v2/app/src/main/java/com/example/lo52_project_v2/model/dao/ParticipantDao.java
package com.example.lo52_project_v2.model.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Delete;
import androidx.room.Transaction;
import androidx.room.Update;

import com.example.lo52_project_v2.model.bo.Participant;
import com.example.lo52_project_v2.model.bo.EquipeParticipants;
import com.example.lo52_project_v2.model.bo.ParticipantEquipe;

import java.util.List;


@Dao
public interface ParticipantDao {
    @Insert
    public void insertParticipant(Participant participant);

    @Query("SELECT * FROM participant")
    List<Participant> getAll();


    @Transaction
    @Query("SELECT * From Equipe where idEquipe LIKE :idEquipe")
    EquipeParticipants getCorrespondingParticipants(int idEquipe);

    @Query("SELECT * From ParticipantEquipe where idEquipe LIKE :idEquipe ")
    List<ParticipantEquipe> getCorrespondingsParticipantsOrder(int idEquipe);

    @Transaction
    @Query("SELECT * From Equipe INNER JOIN ParticipantEquipe ON Equipe.idEquipe = ParticipantEquipe.idEquipe where Equipe.idEquipe LIKE :idEquipe Order by ParticipantEquipe.ordreDePassage ")
    EquipeParticipants getCorrespondingsParticipantsinOrder(int idEquipe);

    @Update
    public void updateParticipantOrder(ParticipantEquipe ParticipantEquipe);

    @Query("SELECT * FROM participant WHERE nomParticipant LIKE :nomParticipant " +
            " LIMIT 1")
    Participant findByName(String nomParticipant);

    @Query("SELECT * FROM participant WHERE id LIKE :id " +
            " LIMIT 1")
    Participant findById(int id);

    @Update
    public void updateParticipant(Participant participant);

    @Delete
    public void deleteParticipant(Participant participant);
}

File: Projet/GestionCourse/Gestion-Course/app/src/main/java/com/example/gestion_course/dao/ParticipantDao.kt
package com.example.gestion_course.dao

import androidx.room.*
import com.example.gestion_course.entities.Participant


@Dao
interface ParticipantDao {
    @Query("SELECT * FROM participant")
    fun getAll(): List<Participant>

    @Query("SELECT * FROM participant WHERE num_equipe_participant = :numEquipe")
    fun getParticipantsParEquipe(numEquipe: Int): List<Participant>

    @Query("SELECT * FROM participant WHERE num_participant IN (:numParticipant)")
    fun loadAllByIds(numParticipant: IntArray): List<Participant>

    @Query("SELECT * FROM participant WHERE num_participant = :numParticipant")
    fun getParticipantById(numParticipant: Int): Participant

    @Update
    fun updateList(participants: List<Participant?>?): Int

    @Query("UPDATE Participant SET num_etape_participant = :numEtape WHERE num_participant = :id")
    fun updateParticipant(id: Int, numEtape: Int)

    @Insert
    fun insertAll(vararg participants: Participant)

    @Delete
    fun delete(participant: Participant)
}